### ISSUE TITLE
Server connection over domain instead of ip fails if ip changed

### ISSUE THREAD

**Issue by Hackebein** [SpigotMC/BungeeCord#1619]
Excerpt from my config.yml (Test System):
```yaml
servers:
  lobby:
    motd: '&1Server: Lobby'
    address: lobby:25565
    restricted: false
  test1:
    motd: '&1Server: test1'
    address: test1:25565
    restricted: false
  test2:
    motd: '&1Server: test2'
    address: test2:25565
    restricted: false
```
Reproduction (Linux):
- Start a game server (spigot)
- Wear the IP of the Games server in your DNS
- Start BungeeCord
- Change the IP of the game server
- Update the entry in your DNS
- BungeeCord tries to establish on the old IP of the server to connect games

Try and Errors:
`-Dsun.net.inetaddr.ttl=0` as start param and the property `networkaddress.cache.ttl=0` in my `lib/security/java.security` doesn't fix it.

Temporary Troubleshooting:
Reload (/greload) the configuration fix it. The Problem now is the red warning and not supportet issues.

My guess:
I suspect that the problem when loading the configuration is. After loading the configuration the domain is resolved only once. And then continues to work only with the IP.

BungeeCord Version: Last stable Build [Build 1088 (Oct 14, 2015 4:10:08 PM)]

**Comment by md-5** [comment]
Wontfix: Addresses are stored by InetSockAddress on creation. greload is the correct way to refresh this part of the configuration.

**Comment by Hackebein** [comment]
You mean that is reload the configuration without making any changes to the configuration is a supported Senario?

**Comment by Hackebein** [comment]
I have further debugged. The result is a non-considered happens here. I will probably submit a patch in the coming days for this purpose.

**Comment by kamcio96** [comment]
Who update dns after server start? Just use ip.

**Comment by Hackebein** [comment]
The question is not "Who update dns after server start?" The fact is, it is possible!

Right now, i have 3 scenarios:
- Long uptime for Bungeecord
- DDNS
- Docker

And i think there are many more ...

**Comment by kamcio96** [comment]
Why don't you use ip in config? It is faster

**Comment by Hackebein** [comment]
because i can't use this in my scenario and it's more flexible.

This tiny little speed disadvantage have any software. It's an advantage and disadvantage discussion. 
Why are you calling a web site by domain instead of an IP?

**Comment by schneidermr** [comment]
Is there any workaround to solve like these sorts of problems? In visualized environments, there isn't any way to fix a container's/pod's IP. And that's a hectic approach to always reload/restart the proxy...
Going through the Bungee source, [I noticed it](https://github.com/SpigotMC/BungeeCord/blob/master/proxy/src/main/java/net/md_5/bungee/conf/Configuration.java#L121) isn't a resolution for this to remove the server and re-add it because there isn't any way to do it...
Then how the hell would be able to solve this problem?

**Comment by Buh13246** [comment]
I'm very sad that Bungeecord saves servers internally as SocketAddresses. I have build a Minecraft Proxy which uses the Server-Address in the Handshake package to route internally. We thought about using bungeecord in front of the proxy which would have worked if the server-Address would be set correctly. So I only get this inside the Proxy:
`ClientToServer: 193.41.226.182:25665 -> [::1]:25566: common.Package{Len:1234, Type:"Handshake", Data:server.HandshakeData{Version:760, ServerAddress:"dev.*****org\x0095.**.***.135\x002b33fd8956ca4f77**eb44b95198014d\x00[{\"name\":\"textures\",\"value\":\"ewogICJ0aW1lc3RhbXAiIDogMTY2ODczMjA0NDAxNCwKICZUlkIiA6ICIyYjMzZmQ4OTU2Y2E0Zjc3YmVlYjQ0Yjk1MTk4MDE0ZCIsCiAgInByb2ZpbGVOYW1lIiA6ICIyMFBoaWxpcHAxMyIsCiAgInNpZ25hdHVyZVJlcXVpcmVkIiA6IHRydWUsCiAgInRleHR1cmVzIiA6IHsKICAgICJTS0lOIiA6IHsKICAgICAgInVybCIgOiAiaHR0cDovL3RleHR1cmVzLm1pbmVjcmFmdC5uZXQvdGV4dHVyZS85YWQ3ZThlN2FkMzU5ZTA3ZDdjZTU0NjdiZWQ4Nzc2ODM4Y2Q3MjMzOGI5NDk2ZDMzMWJlMmMwOTAwMTQ0MzEwIgogICAgfQogIH0KfQ\\u003d\\u003d\",\"signature\":\"Rcqj0PB33/O9MgI4LQhhp9XrnwWBEciw4xO10+WA7Fe0rh8qS785lHG/LVjcfZDTYTkjT7e2lC8PVGUA9NSpglKyT7LDAccucvyAlGiDKnfsUuyskNbNG4ebU3cqCktnrYTdwC77qGteDuS52KvLHOhM0jAoTxxK/KWKloj4iI4T/AmmnlPwHW+r95adwm51/ivXYGvkrYBLSyABrpR5QUxKuZ1N7oKDBwKZWQSuv9J05YSUoKTP0hqGkX8Oc8C/sINNMq74avtK49UvJGR4f8NWHNjsD8RUg30csY6ZqTEa3M8jOA/5KOuaEYUOPOqKvlwIkEKzAU67I6U5McHNZuyYvmAIXAky6Z18NvVheKzGOgoAZNBWZ4YmCrgdf7A2uWK9djFQWTJAUt39rmZH1ooQQkfGV3qjy/vFfUl0x4jxg2VgRkx93MEh5AO0g32m7J0abifjiUstvHarC0uPmnDYD8AgXWHqMfZRFeSj3jAd0o4pcskD3waL368NUTKcxH0GLMLMmq2jlpyMgzOA6xiGg8or9+gtkwxlQnr2v567CEqpQXosrrFHZXPu9RyLT8hdt293WCol+C1N18Cm0DohJ29+P33SHhdYxS5EmEOg8FSTk1z7eLTTfKSKatADN6M5KhBodXakIFwuqJ8BZM2Og5qg\\u003d\"}]", ServerPort:25577, NextState:2}}`

My only two options are:
1. Fork Bungeecord and patch it to store the domain / set ServerAddress correctly
2. Reimplement the switchServer logic inside myProxy and throw Bungeecord away

I really like bungeecord but rewriting the ServerAddress / refetch the DNS entry on the fly would make it a lot better. Is there a technical reason why it was implemented this way?

**Comment by Janmm14** [comment]
@Buh13246 Why you ever need a proxy behind a proxy in the first place??? Throw "myProxy" away and fork bungee, the change is small. Then you use bungee plugin to route whatever you need to route.

**Comment by Buh13246** [comment]
@Janmm14  Sorry you don't know what "myProxy" exactly does. I fixed it, by patching Bungeecord to communicate with my own APIs. My Proxy is a go-based TCP proxy with a lot of custom routing and keeps the TCP Connections alive even when restarting the proxy. So I'm able to live-patch the binary without the clients disconnecting. That is something, that Bungeecord is not Able to do. Also I don't like programming in Java.
